rules.py

```diff
@@ -216,31 +219,84 @@ class ElectoralDistrictOcdId(base.BaseRule):
     )
-
+    github_repo = None
+    
     def __init__(self, election_tree, schema_file):
         super(ElectoralDistrictOcdId, self).__init__(election_tree, schema_file)
+        g = Github()
+        self.github_repo = g.get_repo("opencivicdata/ocd-division-ids")
         self.ocds = self._get_ocd_data()
         self.gpunits = []
         for gpunit in self.election_tree.iterfind("//GpUnit"):
             self.gpunits.append(gpunit)
+            
+    def _get_latest_commit_date(self):
+        """Returns the latest commit date to country-us.csv."""
+        latest_commit_date = None
+        latest_commit = self.github_repo.get_commits(
+            path="/identifiers/country-us.csv")[0]
+        latest_commit_date = latest_commit.commit.committer.date
+        return latest_commit_date
+        
+    def _get_latest_file_blob_sha(self):
+        """Returns the gihub blob sha of country-us.csv."""
+        blob_sha = None
+        dir_contents = self.github_repo.get_dir_contents("identifiers")
+        for content_file in dir_contents:
+            if content_file.name == "country-us.csv":
+                blob_sha = content_file.sha
+                break
+        return blob_sha
+
+    def _download_data(self, file_path):
+        """Makes a request to Github to download the file."""
+        r = requests.get(self._OCDID_URL)
+        with io.open(file_path, "wb") as fd:
+            for chunk in r.iter_content():
+                fd.write(chunk)
+
+    def _verify_data(self, file_path):
+        """Compares blob sha to gihub sha and returns set of ocd id codes
+        if the file is valid
+        """
+        file_sha1 = hashlib.sha1()
+        ocd_id_codes = set()
+        file_info = os.stat(file_path)
+        #github calculates the blob sha like this
+        #sha1("blob "+filesize+"\0"+data)
+        file_sha1.update(b"blob %d\0" % file_info.st_size)
+        with io.open(file_path, mode="rb") as fd:
+            for line in fd:
+                file_sha1.update(line)
+                if line is not "":
+                    ocd_id_codes.add(line.split(",")[0])
+        latest_file_sha = self._get_latest_file_blob_sha()
+        if latest_file_sha != file_sha1.hexdigest():
+            return None
+        else:
+            return ocd_id_codes
 
     def _get_ocd_data(self):
-        """Checks if OCD file is in the curent directory, dowloads it if not."""
+        """Checks if OCD file is in ~/cache, downloads it if not."""
         cache_directory = os.path.expanduser("~/.cache")
         countries_file = cache_directory + "/country-us.csv"
         if not os.path.exists(countries_file):
             if not os.path.exists(cache_directory):
                 os.makedirs(cache_directory)
-            ocd_file = io.open(countries_file, "w")
-            data = requests.get(self._OCDID_URL).text
-            ocd_file.write(data)
-            ocd_file.close()
+            self._download_data(countries_file)
```

Comment:
I'd recommend
1) downloading this to country-us.csv.tmp;
2) verifying that the .tmp file is non-zero size;
3) verifying that the .tmp file matches the Github SHA; and then
4) renaming the .tmp file to country-us.csv

This would ensure that we're not overwriting a potentially-good csv.